test/integration/targets/aws_s3/tasks/main.yml

```diff
@@ -252,3 +252,76 @@
     that:
       - result.changed == False
 # ============================================================
+- name: create a tempfile for multipart test
+  tempfile:
+  register: tmp1
+
+- name: make tempfile 4 GB
+  command:
+    _raw_params: "dd if=/dev/zero of={{ tmp1.path }} bs=1m count=4096"
+
+- name: make a bucket to upload the file
+  aws_s3:
+    bucket: "{{ bucket_name }}"
+    mode: create
+    security_token: "{{security_token}}"
+    aws_access_key: "{{ ec2_access_key }}"
+    aws_secret_key: "{{ ec2_secret_key }}"
+
+- name: upload the file to the bucket
+  aws_s3:
+    bucket: "{{ bucket_name }}"
+    mode: put
+    src: "{{ tmp1.path }}"
+    object: multipart.txt
+    security_token: "{{security_token}}"
+    aws_access_key: "{{ ec2_access_key }}"
+    aws_secret_key: "{{ ec2_secret_key }}"
+
+- name: download file once
+  aws_s3:
+    bucket: "{{ bucket_name }}"
+    mode: get
+    dest: /tmp/multipart_download.txt
+    object: multipart.txt
+    overwrite: different
+    security_token: "{{security_token}}"
+    aws_access_key: "{{ ec2_access_key }}"
+    aws_secret_key: "{{ ec2_secret_key }}"
+  retries: 3
+  delay: 3
+  until: "result.msg == 'GET operation complete'"
+  register: result
+
+- name: assert the file was downloaded once
+  assert:
+    that:
+      - result.changed
+
+- name: download file again
+  aws_s3:
+    bucket: "{{ bucket_name }}"
+    mode: get
+    dest: /tmp/multipart_download.txt
+    object: multipart.txt
+    overwrite: different
+    security_token: "{{security_token}}"
+    aws_access_key: "{{ ec2_access_key }}"
+    aws_secret_key: "{{ ec2_secret_key }}"
+  register: result
```

Comment:
The bucket is never deleted after this test, it looks like.

Reply:
Oops.